crates/modelardb_storage/src/lib.rs

```diff
@@ -311,13 +321,264 @@ fn apache_parquet_writer_properties(
+/// could not be converted to bytes or the record batch could not be created, return
+/// [`ModelarDbStorageError`].
+pub fn model_table_metadata_record_batch(
+    model_table_metadata: &ModelTableMetadata,
+) -> Result<RecordBatch> {
+    // Since the model table metadata does not include error bounds for the generated columns,
+    // lossless error bounds are added for each generated column.
+    let mut error_bounds_all = Vec::with_capacity(model_table_metadata.query_schema.fields().len());
+
+    let lossless = ErrorBound::try_new_absolute(0.0)?;
+
+    for field in model_table_metadata.query_schema.fields() {
+        if let Ok(field_index) = model_table_metadata.schema.index_of(field.name()) {
+            error_bounds_all.push(model_table_metadata.error_bounds[field_index]);
+        } else {
+            error_bounds_all.push(lossless);
+        }
+    }
+
+    let query_schema_bytes = try_convert_schema_to_bytes(&model_table_metadata.query_schema)?;
+    let error_bounds_array = error_bounds_to_list_array(error_bounds_all);
+    let generated_columns_array =
+        generated_columns_to_list_array(model_table_metadata.generated_columns.clone());
+
+    RecordBatch::try_new(
+        TABLE_METADATA_SCHEMA.0.clone(),
+        vec![
+            Arc::new(StringArray::from(vec!["model"])),
+            Arc::new(StringArray::from(vec![model_table_metadata.name.clone()])),
+            Arc::new(BinaryArray::from_vec(vec![&query_schema_bytes])),
+            Arc::new(error_bounds_array),
+            Arc::new(generated_columns_array),
+        ],
+    )
+    .map_err(|error| error.into())
+}
+
+/// Convert a list of [`ErrorBounds`](ErrorBound) to a [`ListArray`].
+fn error_bounds_to_list_array(error_bounds: Vec<ErrorBound>) -> ListArray {
+    let mut error_bounds_builder = ListBuilder::new(Float32Builder::new());
+
+    for error_bound in error_bounds {
+        match error_bound {
+            ErrorBound::Absolute(value) => {
+                error_bounds_builder.values().append_value(value);
+            }
+            ErrorBound::Relative(value) => {
+                // Relative error bounds are encoded as negative values for simplicity.
+                error_bounds_builder.values().append_value(-value);
+            }
+        }
+    }
+
+    error_bounds_builder.append(true);
+    error_bounds_builder.finish()
+}
+
+/// Convert a list of optional [`GeneratedColumns`](GeneratedColumn) to a [`ListArray`].
+fn generated_columns_to_list_array(generated_columns: Vec<Option<GeneratedColumn>>) -> ListArray {
+    let mut generated_columns_builder = ListBuilder::new(StringBuilder::new());
+
+    for generated_column in generated_columns {
+        if let Some(generated_column) = generated_column {
+            if let Some(sql_expr) = generated_column.original_expr {
+                generated_columns_builder.values().append_value(sql_expr);
+            } else {
+                generated_columns_builder.values().append_null();
```

Comment:
If a generated column exists for one table in the cluster will it not created issues if it does not exist for other instances of the table on other nodes in the same cluster? Seems generated columns for which the original expression cannot be extracted should be rejected when the table is created as they cannot be propagated to other nodes in the cluster.

Reply:
I am also not entirely sure why `original_expr` in `GeneratedColumn` is `Option<String>` and not just `String`. If we want to change this, I can do this in a separate PR since I do not think it is relevant to this PR.

Reply:
Yes let us make the change in a separate PR, just resolve the conversation when an issue has been created.